src/components/ui/coaching-sessions/coaching-notes.tsx

```diff
@@ -120,63 +50,142 @@ const CoachingNotes = () => {
 
   if (isLoading) {
     return (
-      <div className="flex flex-col items-center justify-center space-y-4 p-8 min-h-[440px] lg:min-h-[440px] sm:min-h-[200px] md:min-h-[350px]">
-        <div className="w-full max-w-md">
-          <div className="flex items-center justify-between mb-2">
-            <span className="text-sm font-medium text-gray-700 dark:text-gray-300">
-              Loading coaching notes...
-            </span>
-            <span className="text-sm text-gray-500 dark:text-gray-400">
-              {Math.round(loadingProgress)}%
-            </span>
+      <div className="coaching-notes-editor">
+        <div className="coaching-notes-loading">
+          <div className="w-full max-w-md">
+            <div className="flex items-center justify-between mb-2">
+              <span className="text-sm font-medium">
+                Loading coaching notes...
+              </span>
+              <span className="text-sm opacity-70">
+                {Math.round(loadingProgress)}%
+              </span>
+            </div>
+            <Progress value={loadingProgress} className="h-2" />
           </div>
-          <Progress value={loadingProgress} className="h-2" />
         </div>
       </div>
     );
   }
 
-  if (isError) {
+  if (error) {
     return (
-      <div>
-        We could not retrieve your coaching notes. Please try again later.
+      <div className="coaching-notes-editor">
+        <div className="coaching-notes-error">
+          <div className="text-center">
+            <p className="mb-2">⚠️ Could not load coaching notes</p>
+            <p className="text-sm opacity-80">
+              {error.message || 'Please try again later or contact support if the issue persists.'}
+            </p>
+          </div>
+        </div>
       </div>
     );
   }
 
+  // Show cached editor once ready
+  if (isReady && activeExtensions.length > 0) {
+    try {
+      if (process.env.NODE_ENV === 'development') {
+        console.log('🚀 Rendering cached editor with extensions:', activeExtensions.length);
+      }
```

Comment:
Is this needed anymore? Checking for environment in the code like this becomes noisy in my experience. If we want to have something similar to Rust's `debug!` maybe we can develop a module for that that hides these implementation details as part of another task?